ambry-rest/src/test/java/com.github.ambry.rest/FaultyFactory.java

```diff
@@ -29,18 +29,18 @@ public FaultyFactory(Object obj1, Object obj2) {
     // don't care.
   }
 
-  // for RestRequestHandlerFactory and RouterFactory
+  // for RestRequestHandlerFactory
   public FaultyFactory(Object obj1, Object obj2, Object obj3) {
     // don't care.
   }
 
-  // for NioServerFactory
-  public FaultyFactory(Object obj1, Object obj2, Object obj3, Object obj4, Object obj5) {
+  // for BlobStorageServiceFactory and RouterFactory
+  public FaultyFactory(Object obj1, Object obj2, Object obj3, Object obj4) {
     // don't care.
   }
 
-  // for BlobStorageServiceFactory
-  public FaultyFactory(Object... objs) {
+  // for NioServerFactory
+  public FaultyFactory(Object obj1, Object obj2, Object obj3, Object obj4, Object obj5, Object obj6) {
```

Comment:
Constructors with varargs actually don't work with `Utils.getObj()` because it shows up as one array type argument. I will look for a way to fix this.